ding/model/common/head.py

```diff
@@ -536,6 +536,113 @@ def forward(self, x: torch.Tensor) -> Dict:
+            - norm_type (:obj:`str`):
+                The type of normalization to use, see ``ding.torch_utils.fc_block`` for more details
+            - noise (:obj:`bool`): Whether use noisy ``fc_block``
+        """
+        super(StochasticDuelingHead, self).__init__()
+        if a_layer_num is None:
+            a_layer_num = layer_num
+        if v_layer_num is None:
+            v_layer_num = layer_num
+        layer = NoiseLinearLayer if noise else nn.Linear
+        block = noise_block if noise else fc_block
+        self.A = nn.Sequential(
+            MLP(
+                hidden_size + action_shape,
+                hidden_size + action_shape,
+                hidden_size + action_shape,
+                a_layer_num,
+                layer_fn=layer,
+                activation=activation,
+                norm_type=norm_type
+            ), block(hidden_size + action_shape, output_size)
+        )
+        self.V = nn.Sequential(
+            MLP(
+                hidden_size,
+                hidden_size,
+                hidden_size,
+                v_layer_num,
+                layer_fn=layer,
+                activation=activation,
+                norm_type=norm_type
+            ), block(hidden_size, 1)
+        )
+
+    def forward(self, s: torch.Tensor, a: torch.Tensor, mu_t: torch.Tensor, sigma_t: torch.Tensor,
+                sample_size: int = 10) -> Dict:
+        r"""
+        Overview:
+            Use encoded embedding tensor to predict Dueling output.
+            Parameter updates with DuelingHead's MLPs forward setup.
+        Arguments:
+            - s (:obj:`torch.Tensor`):
+                The encoded embedding state tensor, determined with given ``hidden_size``, i.e. ``(B, N=hidden_size)``.
+            - a (:obj:`torch.Tensor`):
+                The encoded embedding action tensor, determined with ``action_size``, i.e. ``(B, N=action_size)``.
+            - mu_t (:obj:`torch.Tensor`):
+                The mu output of actor head at timestep t, size (B, action_size)
+            - sigma_t (:obj:`torch.Tensor`):
+                The sigma output of actor head at timestep t, size (B, action_size)
+            - sample_size (:obj:`int`):
+                The num to samples for action when computing the Q value
+        Returns:
+            - outputs (:obj:`Dict`):
+                Run ``MLP`` with ``DuelingHead`` setups and return the result prediction dictionary.
+                Necessary Keys:
+                    - pred (:obj:`torch.Tensor`): Pred tensor of size ``(B, 1)``.
+        """
+        batch_size = s.shape[0]  # batch size * T
+        hidden_size = s.shape[1]
+        action_size = a.shape[1]  # TODO(pu): action dim is too small than hidden size
+        state_cat_action = torch.cat((s, a), dim=1)  # size (B, action_size + state_size)
+        a_val = self.A(state_cat_action)  # size (B, 1)
+        s_val = self.V(s)  # size (B,1)
+
+        mu_t = (torch.unsqueeze(mu_t, 1)).expand(
```

Comment:
follow the standard distribution sample in PyTorch, you can refer to this [blog](https://bochang.me/blog/posts/pytorch-distributions/)